crates/ty_python_semantic/src/types.rs

```diff
@@ -2355,9 +2358,12 @@ impl<'db> Type<'db> {
 
             // Other than the special cases enumerated above, `Instance` types and typevars are
             // never subtypes of any other variants
-            (Type::NominalInstance(_) | Type::NonInferableTypeVar(_), _) => {
+            (Type::TypeVar(bound_typevar), _) => {
+                // All inferable cases should have been handled above
+                debug_assert!(!inferable.is_inferable(db, bound_typevar));
```

Comment:
what happens if we fail to uphold this invariant... would that lead to very strange diagnostic messages later down the line? If so, might it be worth asserting this in release builds too? Or would that be prohibitively expensive?

```suggestion
                // All inferable cases should have been handled above
                assert!(!inferable.is_inferable(db, bound_typevar));
```

Reply:
Should not be too expensive — this was previously a case that the compiler rejected as not representable, so it shouldn't occur in practice. I'll change to a full assert as you suggest